src/thumbnailer_slope_optim.cc

```diff
@@ -131,36 +122,28 @@ Thumbnailer::Status Thumbnailer::LossyEncodeSlopeOptim(
   while (min_quality <= max_quality && !optim_list.empty()) {
     int mid_quality = (min_quality + max_quality) / 2;
 
-    std::queue<int> new_optim_list;
-
-    while (!optim_list.empty()) {
-      int curr_frame = optim_list.front();
-      optim_list.pop();
+    std::vector<int> new_optim_list;
 
+    for (int& curr_frame : optim_list) {
       float curr_slope;
-      if (ComputeSlope(curr_frame, min_quality, max_quality, &curr_slope) !=
-          kOk) {
-        return kSlopeOptimError;
-      }
+      CHECK_THUMBNAILER_STATUS(
+          ComputeSlope(curr_frame, min_quality, max_quality, &curr_slope));
 
       if (frames_[curr_frame].final_quality == -1 || curr_slope > limit_slope) {
         frames_[curr_frame].config.quality = mid_quality;
-        new_optim_list.push(curr_frame);
+        new_optim_list.push_back(curr_frame);
       }
     }
 
     if (new_optim_list.empty()) break;
 
-    if (GenerateAnimationNoBudget(&new_webp_data) != kOk) {
-      return kMemoryError;
-    }
+    CHECK_THUMBNAILER_STATUS(GenerateAnimationNoBudget(&new_webp_data));
 
     if (new_webp_data.size <= byte_budget_) {
-      while (!new_optim_list.empty()) {
-        int curr_frame = new_optim_list.front();
-        new_optim_list.pop();
+      optim_list.clear();
+      for (int& curr_frame : new_optim_list) {
         frames_[curr_frame].final_quality = mid_quality;
-        optim_list.push(curr_frame);
+        optim_list.push_back(curr_frame);
```

Comment:
In both branches of the if, this ends up being equivalent to optim_list = new_optim_list, right? It would be clearer to have it written this way

Reply:
That's right, I will remove lines 143, 146 and then write `optim_list = new_optim_list;` right below the end of the `if`.